2.train-VAE/optimize_utils_tcvae.py

```diff
@@ -0,0 +1,205 @@
+    )
+    parser.add_argument(
+        "--min_epochs",
+        default=5,
+        type=int,
+        help="Minimum epochs",
+    )
+    parser.add_argument(
+        "--max_epochs",
+        default=1000,
+        type=int,
+        help="Maximum epochs",
+    )
+    parser.add_argument(
+        "--epoch_step",
+        default=100,
+        type=int,
+        help="Epoch step size",
+    )
+    parser.add_argument(
+        "--min_batch_size",
+        default=16,
+        type=int,
+        help="Minimum batch size",
+    )
+    parser.add_argument(
+        "--max_batch_size",
+        default=112,
+        type=int,
+        help="Maximum batch size",
+    )
+    parser.add_argument(
+        "--batch_size_step",
+        default=32,
+        type=int,
+        help="Batch size step size",
+    )
+    parser.add_argument(
+        "--min_lr",
+        default=1e-6,
+        type=float,
+        help="Minimum learning rate",
+    )
+    parser.add_argument(
+        "--max_lr",
+        default=1e-4,
+        type=float,
+        help="Maximum learning rate",
+    )
+    parser.add_argument(
+        "--min_alpha",
+        default=0,
+        type=float,
+        help="Minimum alpha",
+    )
+    parser.add_argument(
+        "--max_alpha",
+        default=5.0,
+        type=float,
+        help="Maximum alpha",
+    )
+    parser.add_argument(
+        "--min_gamma",
+        default=0,
+        type=float,
+        help="Minimum gamma",
+    )
+    parser.add_argument(
+        "--max_gamma",
+        default=5.0,
+        type=float,
+        help="Maximum gamma",
+    )
+    parser.add_argument(
+        "--architecture", 
+        default="onelayer", 
+        help="VAE architecture")
+
+    parser.add_argument(
+        "--dataset", 
+        default="cell-painting", 
+        help="cell-painting or L1000 dataset"
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def objective_tc(trial, train_tensor, val_tensor, train_df):
+    """
+    Optuna objective function
+    Args:
+        trial: Optuna trial
+        train_tensor: Training data tensor
+        val_tensor: Validation data tensor
+        train_df: Training dataframe
+
+    Returns:
+        Validation loss
+    """
+    val_loss = []
+    args = get_optimize_args_tc()
+    """
+    Optuna objective function: optimized by study
+    """
+    # Define hyperparameters
+    latent_dim = trial.suggest_int(
+        "latent_dim", args.min_latent_dim, args.max_latent_dim
+    )
+    beta = trial.suggest_float(
+        "beta", args.min_beta, args.max_beta
+    )
+    learning_rate = trial.suggest_float(
+        "learning_rate", args.min_lr, args.max_lr
+    )
+    batch_size = trial.suggest_int(
+        "batch_size", args.min_batch_size, args.max_batch_size
+    )
+    epochs = trial.suggest_int(
+        "epochs", args.min_epochs, args.max_epochs
+    )
+    optimizer_type = trial.suggest_categorical(
+        "optimizer_type", ["adam","rmsprop"]
```

Comment:
No SGD?

Reply:
just not giving any sort of good results and adjusting other hyperparams to fix has negative effects on adam and rmsprop